enterprise/server/remote_execution/dirtools/dirtools.go

```diff
@@ -928,6 +945,24 @@ func DownloadTree(ctx context.Context, env environment.Env, instanceName string,
 		for _, symlinkNode := range dir.GetSymlinks() {
 			nodeAbsPath := filepath.Join(parentDir, symlinkNode.GetName())
 			if err := os.Symlink(symlinkNode.GetTarget(), nodeAbsPath); err != nil {
+				if os.IsExist(err) {
+					// If the symlink exists and is correct;
+					// skip the error.
+					if checkSymlink(symlinkNode.GetTarget(), nodeAbsPath) {
+						continue
+					}
+					// If the symlink exists but was
+					// incorrect; try to remove it
+					if err := removeSymlink(nodeAbsPath); err != nil {
+						return err
+					}
```

Comment:
wdyt about replacing these two things w/ `os.RemoveAll(nodeAbsPath)`? This way even if the existing path is not a symlink we can just obliterate it and place the correct symlink on top of it.

Reply:
even if a directory already exists at the path, we ultimately want to replace it with a symlink, so I think RemoveAll() is probably the right thing to do.

Also I wasn't sure so had to double check - if the existing path is a symlink to another directory, RemoveAll() does not remove the directory that it points to. It just removes the link.

Reply:
yeah same haha just double checked that. anyway this is Done.